src/Symfony/Component/Messenger/Command/FailedMessagesRemoveCommand.php

```diff
@@ -69,6 +70,24 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $shouldDeleteAllMessages = $input->getOption('all');
 
         $idsCount = \count($ids);
+
+        if (!$receiver instanceof ListableReceiverInterface) {
+            throw new RuntimeException(\sprintf('The "%s" receiver does not support removing specific messages.', $failureTransportName));
+        }
+
+        if (!$idsCount && null !== $input->getOption('class-filter')) {
+            $ids = $this->getMessageIdsByClassFilter($input->getOption('class-filter'), $receiver);
+            $idsCount = \count($ids);
+
+            if (!$idsCount) {
+                throw new RuntimeException('No failed messages were found with this filter.');
+            }
+
+            if (!$io->confirm(\sprintf('There is %d messages to remove. Do you want to continue?', $idsCount))) {
```

Comment:
The message should be `There is 1 message` when there is only one and `There are %d messages` when there are more